docs/jax.config.rst

```diff
@@ -0,0 +1,20 @@
+.. currentmodule:: jax
+
+JAX configuration
+=================
+
+.. autosummary::
+   :toctree: _autosummary
+
+   config
+   enable_checks
+   check_tracer_leaks
+   checking_leaks
+   enable_custom_prng
```

Comment:
This one and `default_prng_impl` might be transient so that we don't want to document them... I'm not sure. @froystig wdyt?

Reply:
If that's the case, we should also avoid declaring them in the JAX namespace (accomplished by importing `foo` rather than importing `foo as foo`) https://github.com/google/jax/blob/5084a12034e6e88cd1fe274f78a9f58e2d109eeb/jax/__init__.py#L40-L52

Reply:
I expect `default_prng_impl` to remain indefinitely. It indicates which PRNG implementation to use if none are specified, and we plan to continue to have multiple implementations. I'd say we keep that one and document it.

`enable_custom_prng` is explicitly intended to be transient (it is an upgrade flag). It is nonetheless public, so the `jax` namespace seems appropriate. Its description text is clear regarding our intent to remove it eventually:

https://github.com/google/jax/blob/3f2a4b10f63c398d5a30a9dd11ffba704de9baf2/jax/_src/config.py#L517-L521

Reply:
Can we reflect the flag's description text in the auto-generated docs view, even if manually?

Reply:
Is this what you have in mind? https://jax--8718.org.readthedocs.build/en/8718/_autosummary/jax.enable_custom_prng.html#jax.enable_custom_prng

Reply:
There it is! Yes.

I was also thinking about the the [`jax.config` page](https://jax--8718.org.readthedocs.build/en/8718/jax.config.html). If we'd like to further distinguish a setting like this one from the others, we could try to add to the first docstring line, so that it shows up on the `jax.config` summary as, say:

> Context manager for _jax_enable_custom_prng_ config option. **[transient]**

rather than the current:

> Context manager for _jax_enable_custom_prng_ config option.

But I'm also happy with things as they are in this PR.

Reply:
That summary is defined here: https://github.com/google/jax/blob/610187cd332ee18b40bc9caf90e54db50f4d0659/jax/_src/config.py#L338

We could modify that for the class in question. Alternatively, we could use a [custom template](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html#customizing-templates) for the autosummary directive, but that might be a too heavy-handed here.

Reply:
We could thread a boolean `transient` flag through all the config definition boilerplate, and then display `[transient]` in the docstring if it's set to True. What do you think?

Reply:
Sounds like a fine plan. Do we have any other transient settings at the moment? @mattjj – any thoughts, given your question that started this thread?

Reply:
Maybe the most flexible approach would be to add an `extra_description` setting that defaults to `""`.

Reply:
Done in #8768